pkg/sql/parser/parse_test.go

```diff
@@ -1720,6 +1720,9 @@ func TestParse(t *testing.T) {
 
 		{`RESTORE TENANT 36 FROM ($1, $2) AS OF SYSTEM TIME '1'`},
 
+		{`RESTORE TENANT 123 FROM REPLICATION STREAM FROM 'bar'`},
+		{`RESTORE TENANT 123 FROM REPLICATION STREAM FROM $1`},
```

Comment:
Let's also add some tests for `RESTORE TABLE foo FROM ...` since it's supported by the grammar. Although we error on it later, I think it's worth noting in these tests that it's a grammar that we support. A comment along with the tests explaining that we'll error out later would be good too.